### PR TITLE
Slightly tighten the Bbox/Transform API.

### PR DESCRIPTION
**Pull request by anntzer**
At least it's explicit that Bbox.inverse_transformed() follows the
Bbox.transformed() semantics, not those of Transform.transform_bbox() or
of TransformedBbox (#12059) (sounds obvious, but you never know...).

Also one less thing to implement for a possible future C implementation of the transform stack.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
I argued in #12059 that the transform API is complex and inconsistent, and would benefit from some tightening.

**Comment by jklymak** [comment]
I'm not following the rationale behind removing this helper....

**Comment by anntzer** [comment]
Do you know the difference between TransformedBbox, Transform.transform_bbox, and Bbox.transformed?  They all do slightly different things (#12059), and I think it's too bad that this is not more obvious from the names of the APIs; in other words there are too many ways to "apply" a transform to a bbox, and I'm trying to start removing some of those.

**Comment by jklymak** [comment]
Agreed its a bit of a mess and cleanup would be great. Do you have a roadmap?  Maybe make a "project" so the rest of us can help/comment/not do contradictory things?  

In the meantime though, this still seems like a useful helper :wink:

**Comment by anntzer** [comment]
It's basically discussed in #12059 -- still in progress.

I would argue that `bbox.transformed(trf.inverted())` is not much more onerous to type than `bbox.inverse_transformed(trf)` (just two more characters), so let's just be explicit...

**Comment by timhoffm** [comment]
Unfortunately, needs rebase again due to the broken imports.

**Comment by anntzer** [comment]
rebased

File: doc/api/next_api_changes/2018-01-06-AL.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Deprecations
```
Comment (tacaswell) [inline doc/api/next_api_changes/2018-01-06-AL.rst]: move to the new scheme.

File: lib/matplotlib/transforms.py [hunk 2]
```diff
@@ -513,6 +513,7 @@ def transformed(self, transform):
             [pts[0, 0], pts[1, 1]], [pts[1, 0], pts[0, 1]]]))
         return Bbox([ll, [lr[0], ul[1]]])
 
+    @cbook.deprecated("3.2", alternative="transformed(transform.inverted())")
```
Comment (tacaswell) [inline lib/matplotlib/transforms.py]:
```suggestion
    @cbook.deprecated("3.3", alternative="transformed(transform.inverted())")
```

Comment (anntzer) [reply lib/matplotlib/transforms.py]: done

File: doc/api/next_api_changes/2018-01-06-AL.rst (hunk 1 above)
Comment (anntzer) [reply doc/api/next_api_changes/2018-01-06-AL.rst]: done

**Comment by efiring** [comment]
As a first step in simplifying the baffling transform API, I would like to see this go in.

**Comment by anntzer** [comment]
rebased

**Pull request merged**